### PR TITLE
Feat/add marl eval json logger

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
A json logger that writes data in json files in the format that is expected for downstream use by `marl-eval`. This logger is adapted from the one found in [BenchMARL](https://github.com/facebookresearch/BenchMARL/blob/31c88e8574a5f4ade7020e195933a6c2300d24da/benchmarl/experiment/logger.py#L283). 

## Extra
Right now, this logger will write separate `json` files for each experiment ie. different seed, environment, task combinations will be in different `json` files in different directories by default. This is on purpose due to the fact that we often times run distributed experiments. A suggestion for now is that we write a helper function that can take in paths to multiple json files and combine them into one. 
In the case where experiments will be run on the same machine, a path can be set in the experiment config which will allow all experiments to write a single json file.

### CODE REVIEW

**Review by OmaymaMahjoub — APPROVED** [review]
Works fine when we test it on `marl-eval` :fire:

File: mava/configs/logger/ff_ippo.yaml [hunk 1]
```diff
@@ -7,3 +7,4 @@ system_name: ff_ippo
 kwargs:
   neptune_project: Instadeep/Mava
   neptune_tag: [ff-ippo, rware]
+  json_path: test_joint_json
```
Comment (OmaymaMahjoub) [inline mava/configs/logger/ff_ippo.yaml:10]: can you add a comment to explain the json_path

**Review by OmaymaMahjoub — APPROVED** [review]
Thanks @RuanJohn for this PR :fire:

**Review by callumtilbury — COMMENTED** [review]
Haven't actually run it, but logic looks good :)

**Review by callumtilbury — DISMISSED** [review]
Gave it a run & it looks good. Just maybe the base logger change, if you feel it's necessary :)

File: mava/configs/logger/ff_ippo.yaml [hunk 2]
```diff
@@ -3,7 +3,9 @@ defaults:
 
 system_name: ff_ippo
 
-# --- Neptune options ---
+# --- Other logger kwargs ---
 kwargs:
   neptune_project: Instadeep/Mava
   neptune_tag: [ff-ippo, rware]
+  json_path: ~ # If set, json files will be logged to a set path so that multiple experiments can
```
[the remark this answers is not in the document]
Comment (RuanJohn) [reply mava/configs/logger/ff_ippo.yaml]: Yes, and this is a much better idea than what I have done here. Changed it.

**Review by callumtilbury — APPROVED** [review]

**Review by OmaymaMahjoub — APPROVED** [review]
Thanks @RuanJohn!

**Pull request merged**